### ISSUE TITLE
Infantry should only walk in straight lines between cells

### ISSUE THREAD

**Issue by tovl** [OpenRA/OpenRA#18292]
In the original games infantry movement was always straight between (sub-)cells. In OpenRA infantry behaves exactly like vehicles in this regard and makes curved paths. This looks a bit weird, especially in isometric mods, partly because infantry only have 8 facings.

Making infantry turn in place for each cell they pass does not significantly slow them down, because they turn instantly anyway. It will help a bit with making infantry more responsive and quicker to stop, because they cannot be in a situation anymore where they have to move an additional cell due to being in the middle of a curve.

The same was true for walker vehicles in the original TS as well, even though they have a finite turn rate and therefore this will make them somewhat slower overall, so I think this is a bit up for debate.

**Comment by pchote** [comment]
The move behaviour isn't ideal for vehicles or aircraft either. This is one area where we would benefit a lot from understanding the behavior implemented in the original RA/TD source code.

**Comment by tovl** [comment]
In general I agree that it is not ideal, which is why I have been working on refactoring the movement logic before. However, that is quite a big task and after spending some time studying the movement in the originals I realized that the original movement logic is quite bad and in some respect what we currently have is already better than the originals. Especially for voxels, our movement looks so much smoother&mdash;the originals never really used the voxels to their full potential.

Also, when talking specifically about aircraft, well I did have a quick look at the source code and that's where I got the two facings idea from in the first place :smile: .

In any case, infantry/walker movement is one of the most simple cases ,which I already have a working prototype for and doesn't depend on more fundamental refactoring. It will also stay a special case even if we rework everything else about movement (unless we change the artwork to have more than 8 facings). So if there is agreement that the original behaviour is indeed better in this case, I will prepare a PR just for infantry movement.
